apps/pie-docs/src/_11ty/shortcodes/usage.js

```diff
@@ -0,0 +1,87 @@
+const pieIconsSvg = require('../filters/pieIconsSvg');
+const pieDesignTokenColours = require('../filters/pieDesignTokenColours');
+const list = require('./list');
+const usageTypes = require('../../_data/usageTypes');
+
+const metadata = {
+    do: {
+        displayName: 'Do',
+        styleColour: 'support-positive',
+        iconFill: 'support-positive',
+        iconName: 'check-circle',
+    },
+    dont: {
+        displayName: "Don't",
+        styleColour: 'support-error',
+        iconFill: 'support-error',
+        iconName: 'close-circle',
+    },
+};
+
+const buildImage = ({
+    width, alt, mobileSrc, src,
+}) => {
+    const isImageFullContainerWidth = !width;
+    const imageStyles = !isImageFullContainerWidth ? `style="--img-width: ${width};"` : ''; // If image isn't full width, set it to required width
+    const imageAlt = `alt="${alt || ''}"`;
+    const mobileImageMaxWidth = '600px';
+
+    return `<figure class="c-usage-img">
+          <picture>
+            ${mobileSrc ? `<source ${imageStyles} media="(max-width: ${mobileImageMaxWidth})" srcset="${mobileSrc}">` : ''}
+            <img src="${src}" ${imageStyles} ${imageAlt}>
+          </picture>
+      </figure>`;
+};
+
+const buildUsageCard = (usageType, { type, items }) => {
+    const {
+        iconName, iconFill, styleColour, displayName,
+    } = metadata[usageType];
+    const isImage = type === usageTypes.image;
+    const styleColourValue = pieDesignTokenColours({ tokenName: styleColour, tokenPath: ['alias', 'default'] });
+    const svg = pieIconsSvg({
+        name: iconName,
+        attrs: {
+            height: 24,
+            width: 24,
+            fill: iconFill,
+        },
+    });
+    const backdropClasses = ['c-usage-backdrop', ...(isImage ? ['c-usage-backdrop--hasImage'] : [])];
+
+    const content = isImage
+        ? items.map((i) => buildImage(i)).join(' ')
+        : list({ type: 'bullet', items });
+
+    return `
+    <div class="c-usage" style="--style-colour: ${styleColourValue};">
+      <div class="c-usage-heading">
+        ${svg}
+        ${displayName}
+      </div>
+      <div class="${backdropClasses.join(' ')}">
+        ${content}
```

Comment:
It doesn't have to be a part of this PR but I think we need some proper `alt` text on these images. 

I don't think someone with a screen reader would know what we're showing here.

<img width="738" alt="Screenshot 2023-08-01 at 10 13 48" src="https://github.com/justeattakeaway/pie/assets/16766185/287becf9-e4ea-454c-8d6b-4be029efb2ad">